parser/xml/src/test/java/org/codice/ddf/parser/xml/TestXmlParser.java

```diff
@@ -0,0 +1,228 @@
+
+        thrown.expect(ClassCastException.class);
+        ChildElement unmarshal = parser.unmarshal(configurator, ChildElement.class, is);
+    }
+
+    @Test
+    public void testBadContextPath() throws Exception {
+        configurator.setContextPath(ImmutableList.of(""));
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        thrown.expect(ParserException.class);
+        parser.marshal(configurator, mother, os);
+    }
+
+    @Test
+    public void testBadMarshal() throws Exception {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        thrown.expect(ParserException.class);
+        parser.marshal(configurator, this, os);
+    }
+
+    @Test
+    public void testBadUnmarshal() throws Exception {
+        ByteArrayInputStream is = new ByteArrayInputStream(new byte[] {0, 1, 2});
+
+        thrown.expect(ParserException.class);
+        ChildElement unmarshal = parser.unmarshal(configurator, ChildElement.class, is);
+    }
+
+    @Test
+    public void testTypeAdapter() throws Exception {
+        // TODO RAP 30 Jun 15: Actually need to *test* the type adapter
+        //        configurator.setAdapter(new XmlAdapter() {
+        //            @Override
+        //            public Object unmarshal(Object v) throws Exception {
+        //                return null;
+        //            }
+        //
+        //            @Override
+        //            public Object marshal(Object v) throws Exception {
+        //                return null;
+        //            }
+        //        });
+        //        parser.marshal(configurator, mother, os);
+        //        String outputXml = os.toString();
+        //
+        //        assertXpathEvaluatesTo("Padme", "/mother/@firstname", outputXml);
+        //
+        //        ByteArrayInputStream is = new ByteArrayInputStream(os.toByteArray());
+        //        MotherElement unmarshal = parser.unmarshal(configurator, MotherElement.class, is);
+        //
+        //        assertEquals(mother.getAge(), unmarshal.getAge());
+    }
+
+    @Test
+    public void testCacheKey() throws Exception {
+        XmlParser.CacheKey cacheKey1 = new XmlParser.CacheKey("hello:world",
+                TestXmlParser.class.getClassLoader());
+        XmlParser.CacheKey cacheKey2 = new XmlParser.CacheKey("here:now",
+                TestXmlParser.class.getClassLoader());
+        XmlParser.CacheKey cacheKey3 = new XmlParser.CacheKey("hello:world",
+                TestXmlParser.class.getClassLoader());
+
+        assertThat(cacheKey1, not(equalTo(cacheKey2)));
+        assertThat(cacheKey1.hashCode(), not(equalTo(cacheKey2.hashCode())));
+
+        assertTrue(cacheKey1.equals(cacheKey3));
+        assertThat(cacheKey1.hashCode(), equalTo(cacheKey3.hashCode()));
+
+        assertThat(cacheKey1, equalTo(cacheKey1));
+        assertThat(cacheKey1, not(equalTo(null)));
+
+        assertEquals(cacheKey1.equals(null), false);
```

Comment:
Appallingly enough, `assertThat` sees the null and doesn't run the equals method. It wants to _help_.